### ISSUE TITLE
add a generic data bucket, so tests may store additional metadata

### ISSUE THREAD

**Issue by catdad** [catdad/grandma#177]
... this is useful when writing other tooling to look at results.

```javascript
module.exports = {
    test: function (done) {
        ...
        this.data.someProp = 'value';
        ...
    }
};
```

- [x] add code and tests
- [x] add README topic
- [ ] publish

**Comment by catdad** [comment]
Note #178. The usage of the `afterAll` function will not be fixed at this time.

**Comment by catdad** [comment]
Published in v0.8.0.
